kedro/framework/cli/utils.py

```diff
@@ -497,3 +497,38 @@ def _split_load_versions(ctx: click.Context, param: Any, value: str) -> dict[str
         load_versions_dict[load_version_list[0]] = load_version_list[1]
 
     return load_versions_dict
+
+
+def validate_input_with_regex_pattern(pattern_name: str, input: str) -> None:
```

Comment:
I think this should still be a private method as we don't want users to start using this.